### DISCUSSION TITLE
Wyze v3 Cam Pro Video Quality

### DISCUSSION THREAD

**Discussion by TheMegamind** [mrlt8/docker-wyze-bridge#611] [Q&A]
I'm running v1.9.0 in a docker container on an RPi 3B+ with a combination of cameras, including the v3 Pro, v3, and a Pan v2.  I've enabled only those three for now, but have other cameras available.  

The snapshots are all three cameras are crystal clear, as are the video feeds from the v3 and Pan v2.  However, the video feed from the Pro version is grainy and quite poor by comparison. I tried enabling a specific quality setting for v3 Pro, but to no effect.  

Suggestions?

**Comment by mrlt8** [comment]
Hmm. Can you see if it's actually pulling the 2K footage? Does it also look noticeably worse than the wyze app?

**Comment by TheMegamind** [comment]
The image quality is noticeably degraded from the Wyze app. I looked under 'camera details' on the browser window, but I'm not sure how to check what it's pulling.   

Snapshot
![Snapshot](https://user-images.githubusercontent.com/12296628/201527604-88521c03-41d9-4efb-bbb6-88ce58489049.png)

Video
![Video](https://user-images.githubusercontent.com/12296628/201527613-1c43ea56-131a-4546-a37e-58b360961508.png)

**Reply by mrlt8** [reply]
hmm, that is strange, the video look like 360p stream.

Does visiting the snapshot endpoint result in the lower quality image? `http://localhost:5000/snapshot/camera-nickname.jpg`

Anything in the logs? Can you post your docker-compose (don't forget to remove your credentials)?

**Reply by TheMegamind** [reply]
My docker-compose is pretty vanilla at this point, but it's posted below.  

The image from the snapshot endpoint image is not degraded like the second image posted above.  

I tried to look for the logs earlier, as my original thought was that it might be a network issue or something with my RPi slowing things down, but I wasn't immediately familiar enough with Docker or Wyze-Bridge to find them. I'll check the documentation, or perhaps you can point me in the right direction.  

version: '2.4'
services:
    wyze-bridge:
        container_name: wyze-bridge
        restart: unless-stopped
        image: mrlt8/wyze-bridge:latest
        ports:
            - 1935:1935 # RTMP
            - 8554:8554 # RTSP
            - 8888:8888 # HLS
            - 5000:5000 # WEB-UI
        environment:
            - WYZE_EMAIL=xxx
            - WYZE_PASSWORD=xxx
            - FILTER_NAMES=xxx
            - FILTER_BLOCK=true
            - ON_DEMAND=True

**Reply by mrlt8** [reply]
hmm. Everything seems fine. I don't know what could be causing the video to degrade. 
Where are you viewing the stream? Could you try opening the rtsp stream in VLC?

**Comment by TheMegamind** [comment]
I tried it via the rtsp stream in VLC and also ran wyze-bridge out of a docker container on Windows 11.  Same issue.

If others are having success, it must be me. 🤷🏻‍♂️

**Reply by mrlt8** [reply]
Very strange! `snapshot/camera-nickname.jpg` _should_ be taking the snapshot from the same RTSP stream, so I can't figure out why the video is getting degraded...

**Reply by TheMegamind** [reply]
Thanks. Appreciate the effort anyway.

**Comment by AngryMooses** [comment]
Did you have any success on this?  I'm debating between the v3 and v3 Pro right now.

**Reply by TheMegamind** [reply]
No. I was happier with the v3.
